js/hub.md

```diff
@@ -16,30 +14,20 @@ import { Hub } from '@aws-amplify/core';
 
 ## Working with the API
 
-### dispatch()
-
-You can dispatch an event with `dispatch` function:
-```javascript
-Hub.dispatch('auth', { event: 'signIn', data: user }, 'Auth');
-```
+### Listening for messages
 
-### listen()
+`Hub.listen(channel: string | RegExp, callback)` is used to listen for messages that have been dispatched. You must provide either a named `channel` or a regular expression, along with a callback. In the case of a regular expression only dispatches which contain a `message` in their payload will be matched against your pattern. You can add multiple listeners to your application for different channels or patterns to listen for, or trap generic events and perform your own filtering.
 
-You can subscribe to a channel with `listen` function:
 ```javascript
-import { Hub, Logger } from 'aws-amplify';
-
-const logger = new Logger('MyClass');
+import { Hub } from 'aws-amplify';
 
 class MyClass {
     constructor() {
-        Hub.listen('auth', this, 'MyListener');
-    }
-
-    // Default handler for listening events
-    onHubCapsule(capsule) {
-        const { channel, payload } = capsule;
-        if (channel === 'auth') { this.onAuthEvent(payload); }
+        Hub.listen('auth', (data) => {
+            const { channel, payload } = data;
+            if (channel === 'auth') { this.onAuthEvent(payload); }            
```

Comment:
checking the `channel` in the callback seems redundant since we are listening to a specific channel here.

Reply:
Good point I can remove, It was in the old docs and just converted.

Reply:
Fixed